dask/dataframe/io/tests/test_parquet.py

```diff
@@ -2093,11 +2097,63 @@ def test_split_row_groups_pyarrow(tmpdir):
     )
 
     ddf3 = dd.read_parquet(
-        tmp, engine="pyarrow", gather_statistics=True, split_row_groups=True
+        tmp,
+        engine="pyarrow",
+        gather_statistics=True,
+        split_row_groups=True,
+        chunksize=1,
     )
     assert ddf3.npartitions == 12
 
     ddf3 = dd.read_parquet(
         tmp, engine="pyarrow", gather_statistics=True, split_row_groups=False
     )
     assert ddf3.npartitions == 4
+
+
+@pytest.mark.parametrize("metadata", [True, False])
+@pytest.mark.parametrize("chunksize", [None, 1024, 4096, "1MiB"])
+def test_chunksize(tmpdir, chunksize, engine, metadata):
+    check_pyarrow()
+
+    df = pd.DataFrame(
+        {
+            "a": np.random.choice(["apple", "banana", "carrot"], size=100),
+            "b": np.random.random(size=100),
+            "c": np.random.randint(1, 5, size=100),
+            "index": np.arange(0, 100),
+        }
+    ).set_index("index")
+
+    nparts = 2
+    row_group_size = 6
+    ddf1 = dd.from_pandas(df, npartitions=nparts)
+    ddf1.to_parquet(
+        str(tmpdir),
+        engine="pyarrow",
+        row_group_size=row_group_size,
+        write_metadata_file=metadata,
+    )
+
+    if metadata:
+        path = str(tmpdir)
+    else:
+        dirname = str(tmpdir)
+        files = os.listdir(dirname)
+        assert "_metadata" not in files
+        path = os.path.join(dirname, "*.parquet")
+
+    ddf2 = dd.read_parquet(
+        path,
+        engine=engine,
+        chunksize=chunksize,
+        split_row_groups=True,
+        gather_statistics=True,
+        index="index",
+    )
+
+    assert_eq(ddf1, ddf2, check_divisions=False)
+
+    for df_part in ddf2.partitions:
+        if chunksize and len(df_part) > row_group_size:
+            assert df_part.compute().memory_usage().sum() <= parse_bytes(chunksize)
```

Comment:
I don't think that this test checks that we actually merge things.  Maybe something like the following would work?

```python
def test_chunksize(tmpdir, engine):
    df = # make a dataframe with 10 small partitions
    df.to_parquet(...)

    df2 = dd.read_parquet(..., chunksize="2 kB")  # or something like this.
    assert df2.npartitions == 2 # or whatever the expected value is
```